src/firmware_new/rules.bzl

```diff
@@ -0,0 +1,2345 @@
+def stm32h7_hal_library_files_genrule(name):
+    """ Downloads and extracts the Drivers / Middleware For The STM32H7 MCU's """
+
+    # First download the firmware. We have this as a seperate genrule in case we need to
+    # debug later rules, don't want to have to re-download every time
+    # TODO: more descriptive name?
+    native.genrule(
+        name = "{}_download_stm32h7_firmware_zip".format(name),
+        outs = ["fw.zip"],
+        cmd = "\n".join([
+            # Download the drivers and middlewares. We got this URL by running wireshark and
+            # watching where STM32CubeMX tried to download files from when generating code
+            "curl -L http://www.ebuc23.com/s3/stm_test/software/firmware/stm32cube_fw_h7_v150.zip -o fw.zip",
+            "mv fw.zip $@",
+        ]),
+    )
+
+    native.genrule(
+        name = "{}_unzip_stm32h7_firmware".format(name),
+        srcs = ["fw.zip"],
+        # We got this list by running `find . -iname "*.c"` and `find . -iname "*.h"` over the
+        # extracted drivers and middleware folders
```

Comment:
:open_mouth: Is there anyway for the genrule to do this? Would we ever want to upgrade drivers?

Reply:
I looked into having the genrule do this (truly, this pains me), but I couldn't come up with anything. If we ever want to upgrade drivers for the stm32h7, the file paths should mostly be the same, but yeah, I agree, this is not ideal.